addons/web/models/models.py

```diff
@@ -213,36 +219,200 @@ def fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu
+                while parent_id:
+                    values = values_range[parent_id]
+                    local_counters[parent_id]
+                    values['count'] += count
+                    parent_id = values[parent_name]
+
+
     @api.model
-    def search_panel_select_range(self, field_name):
+    def search_panel_range(self, field_name, **kwargs):
+        """
+        Return possible values of the field field_name,
+        possibly with counters and other info (if the extra parameter
+        comodel_field_names has been provided).
+
+        :param field_name: the name of a filter field;
+            possible types are many2one, many2many, selection.
+        :param category_domain: domain generated by categories
+                                (this parameter is used in search_panel_local_counters)
+        :param comodel_domain: domain of field values (if relational)
+                                (this parameter is used in search_panel_local_counters)
+        :param comodel_field_names: a dict with keys given by field names of the comodel
+                                    of field_name (if relational).
+                                    the values are functions used to process
+                                    the field names values. They must accept the single
+                                    parameter record (see below)
+        :param disable_counters: whether to count records by value. This is done in
+                                    search_panel_local_counters
+        :param filter_domain: domain generated by filters
+                                (this parameter is used in search_panel_local_counters)
+        :param search_domain: base domain of search
+                                (this parameter is used in search_panel_local_counters)
+        :return: a list of possible values, each being a dict with keys
+            'id' (value),
+            'name' (value label),
+            'count' (how many records with that value),
+            and possibly other keys
         """
-        Return possible values of the field field_name (case select="one")
-        and the parent field (if any) used to hierarchize them.
+        field = self._fields[field_name]
+
+        local_counters = {}
+        if field.type != 'many2many' and not kwargs.get('disable_counters'):
+            local_counters = self.search_panel_local_counters(field_name, **kwargs)
+
+        if field.type in ['many2many', 'many2one']:
+            Comodel = self.env[field.comodel_name]
+            comodel_domain = kwargs.get('comodel_domain', [])
+            field_names = kwargs.get('comodel_field_names', {})
+            # we fetch display_name but does not process it
+            field_names['display_name'] = lambda record: None
+
+            records = Comodel.with_context(hierarchical_naming=False).search_read(comodel_domain, field_names.keys())
+
+            values_range = OrderedDict()
```

Comment:
Since Odoo 13.0, we require Python 3.6 where all dicts are ordered.
So you can safely use `{}` instead of `OrderedDict()`.
The latter is only useful for implementing an LRU cache.

Reply:
ok